packages/node-core/src/configure/migration-service/migration.ts

```diff
@@ -145,47 +280,231 @@ export class Migration {
+          });
+          this.extraQueries.push(
+            commentConstraintQuery(`"${this.schemaName}"."${rel.target.tableName}"`, fkConstraint, tags),
+            createUniqueIndexQuery(this.schemaName, relatedModel.tableName, relation.foreignKey)
+          );
+          break;
+        }
+        case 'hasMany': {
+          const rel = model.hasMany(relatedModel, {
+            foreignKey: relation.foreignKey,
+          });
+          const fkConstraint = getFkConstraint(rel.target.tableName, rel.foreignKey);
+          const tags = smartTags({
+            foreignFieldName: relation.fieldName,
+          });
+          this.extraQueries.push(
+            commentConstraintQuery(`"${this.schemaName}"."${rel.target.tableName}"`, fkConstraint, tags)
+          );
+          break;
+        }
+        default:
+          throw new Error('Relation type is not supported');
+      }
+      const relationQuery = generateForeignKeyStatement(model.getAttributes()[relation.foreignKey], model.tableName);
+      if (relationQuery) {
+        this.mainQueries.push(relationQuery);
+      }
+    }
+
+    this.addModelToSequelizeCache(model);
+  }
+
+  private addRelationComments(): void {
+    this.foreignKeyMap.forEach((keys, tableName) => {
+      const comment = Array.from(keys.values())
+        .map((tags) => smartTags(tags, '|'))
+        .join('\n');
+      const query = commentTableQuery(`"${this.schemaName}"."${tableName}"`, comment);
+      this.extraQueries.push(query);
+    });
+  }
+
+  dropRelation(relation: GraphQLRelationsType): void {
+    if (relation.type !== 'belongsTo') {
+      return;
+    }
+    const tableName = modelToTableName(relation.from);
+    const fkConstraint = getFkConstraint(tableName, relation.foreignKey);
+
+    this.mainQueries.unshift(dropRelationStatement(this.schemaName, tableName, fkConstraint));
+
+    // TODO remove from sequelize model
+    // Should update sequelize model cache
+  }
+
+  async createEnum(e: GraphQLEnumsType): Promise<void> {
+    const queries: string[] = [];
+
+    // Ref: https://www.graphile.org/postgraphile/enums/
+    // Example query for enum name: COMMENT ON TYPE "polkadot-starter_enum_a40fe73329" IS E'@enum\n@enumName TestEnum'
+    // It is difficult for sequelize use replacement, instead we use escape to avoid injection
+    // UPDATE: this comment got syntax error with cockroach db, disable it for now. Waiting to be fixed.
+    // See https://github.com/cockroachdb/cockroach/issues/44135
+    const enumTypeName = enumNameToHash(e.name);
+    let type = `"${this.schemaName}"."${enumTypeName}"`;
+    let [results] = await this.sequelize.query(
+      `SELECT pg_enum.enumlabel as enum_value
+         FROM pg_type t JOIN pg_enum ON pg_enum.enumtypid = t.oid JOIN pg_catalog.pg_namespace n ON n.oid = t.typnamespace
+         WHERE t.typname = ? AND n.nspname = ? order by enumsortorder;`,
+      {replacements: [enumTypeName, this.schemaName]}
+    );
```

Comment:
This is a very similar query to in `getEnumDeprecated` we should try and reuse the same parts of the query